Mathlib/Combinatorics/SimpleGraph/Walk.lean

```diff
@@ -1367,4 +1377,80 @@ theorem map_toDeleteEdges_eq (s : Set (Sym2 V)) {p : G.Walk v w} (hp) :
 
 end Walk
 
+/-! ## Subwalks -/
+
+namespace Walk
+
+variable {V : Type*} {G : SimpleGraph V}
+
+/-- `p.IsSubwalk q` means that the walk `p` is a contiguous subwalk of the walk `q`. -/
+def IsSubwalk {u₁ v₁ u₂ v₂} (p : G.Walk u₁ v₁) (q : G.Walk u₂ v₂) : Prop :=
+  ∃ (ru : G.Walk u₂ u₁) (rv : G.Walk v₁ v₂), q = (ru.append p).append rv
+
+@[refl, simp]
+lemma isSubwalk_rfl {u v} (p : G.Walk u v) : p.IsSubwalk p := by
+  use nil, nil
+  rw [nil_append, append_nil]
```

Comment:
```suggestion
lemma isSubwalk_rfl {u v} (p : G.Walk u v) : p.IsSubwalk p :=
  ⟨nil, nil, by simp⟩
```

I personally find this styling cleaner and easier to read, but I wouldn't insist on changing it. (same for the other cases below of `use` + one tactic)